homeassistant/helpers/entity.py

```diff
@@ -140,8 +121,8 @@ def update_ha_state(self, force_refresh=False):
         if ATTR_UNIT_OF_MEASUREMENT not in attr and self.unit_of_measurement:
             attr[ATTR_UNIT_OF_MEASUREMENT] = self.unit_of_measurement
 
-        if ATTR_HIDDEN not in attr:
-            attr[ATTR_HIDDEN] = bool(self.hidden)
+        if _OVERWRITE_HIDDEN.get(self.entity_id, self.hidden):
+            attr[ATTR_HIDDEN] = True
```

Comment:
This is clever because it will prevent the "hidden: False" message from showing up in all the state cards for sensors and switches on the frontend. I like it.